### ISSUE TITLE
Client cancelling request gets logged as error

### ISSUE THREAD

**Issue by peterklijn** [zalando/skipper#1654]
**Describe the bug**

Whenever a client cancels a request before the backend is able to process it, Skipper logs an error. I would assume that since this is something the client does, it's not really an error from Skipper's perspective. 

Could you elaborate why Skipper logs this as an error here:
https://github.com/zalando/skipper/blob/master/proxy/proxy.go#L1246-L1260

Would a loglevel `info` not make more sense in this case? Or am I missing something?

Example of error:
```
client canceled after 150.438876ms, route name_of_my_route_with_slow_backend with backend network http://service.namespace.svc, status code 499: dialing failed false: context canceled, remote host: 83.x.x.x, request: "POST /some/action HTTP/1.1", user agent: "PostmanRuntime/7.26.8
```

**To Reproduce**

Add context, for example:

Have a route in Skipper with a backend, like this:

```go
*eskip.Route{
	Id:     "some_route_id",
	Method: http.MethodPost,
	Predicates: []*eskip.Predicate{
		{
			Name: zrouting.PathName,
			Args: []interface{}{"/some/action"},
		},
	},
	Backend:     "http://service.namespace.svc",
	BackendType: eskip.NetworkBackend,
}
```
In my case, use a client of sorts, in my case postman, make the call and cancel it.

**Comment by szuecs** [comment]
@peterklijn yes I agree, that it's not always a skipper error, but also GW timeout is not a skipper error (only if skipper is too slow, which is not likely considering to have backends that should do more work than skipper) and GW timeout we also have error logs.

Logs to consider to have `error -> info`:
https://github.com/zalando/skipper/blob/7c1c975916e316a240e397df21eeabaea3218284/proxy/proxy.go#L936
https://github.com/zalando/skipper/blob/7c1c975916e316a240e397df21eeabaea3218284/proxy/proxy.go#L941
https://github.com/zalando/skipper/blob/master/proxy/proxy.go#L1105

I am honestly not sure what to do, do people expect to only log error on skipper errors, how do we classify it?
I often filter by these "errors" to see what kind of endpoints might had issues. Often this can be reduced to only one or two with issues. The log level does not matter to much to me besides it should be visible.

**Comment by peterklijn** [comment]
Thanks for your reply @szuecs :)

Finding the right log level is hard. Personally, `error` is something that requires action, something is really wrong. From what you describe, I interpret it as follows. Seeing one of these logs is fine, nothing is wrong. but when you see a lot of them (compared to an hour before, or compared to other routes with equal traffic), it might require action.
If `info` doesn't have the right weight to it, perhaps `warning` would do.

But more importantly, is this something that can be an actual error in Skipper? I assume it's not, it's always a client cancelling the request before the server responds.

If no right level can be found, making this configurable would also help :) 

WDYT?

**Comment by szuecs** [comment]
If we change anything I would prefer info, because I have a strong bias for warning is only for depreciations.
In general we don’t know if skipper is the major contributor to the client cancellation or not. Normally not, so maybe the change to info is the right to do.
@aryszka @AlexanderYastrebov wdyt?

**Comment by AlexanderYastrebov** [comment]
The `errorResponse` is executed whenever there is a error during proxying request. The case for client cancellation was introduced in #1391 and later extended in #1392 and #1407; before that it was logged as a generic error. I think it can be converted to `Info` or even removed as it is also duplicated in access log:
```
[APP]ERRO[0026] client canceled after 95.622377ms, route  with backend network https://example.com, flow id abc, status code 499: dialing failed false: context canceled, remote host: 127.0.0.1, request: "GET / HTTP/1.1", user agent: "curl/7.58.0" 
127.0.0.1 - - [18/Dec/2020:00:53:19 +0100] "GET / HTTP/1.1" 499 1 "-" "curl/7.58.0" 95 localhost:9090 abc -
```

**Comment by peterklijn** [comment]
I've updated the log level for client canceled errors here:
https://github.com/zalando/skipper/pull/1659

@szuecs I did not update the backend  roundtrip errors as I don't fully understand them.

**Comment by peterklijn** [comment]
Hi @szuecs or anybody else from the team, could you have a look at the PR please?
I would like to get rid of the errors in our production logs :) j

**Comment by aryszka** [comment]
Hi @peterklijn , the PR looks good to me.

I would even consider it to degrade the log level to Debug, because the relevant information at large scale is the number of the 499s, which we can get from the metrics. As a first step, changing to Info is ok, though, if that's more preferred by others.

**Comment by peterklijn** [comment]
I agree @aryszka, I changed it to info as it felt like a 'smaller' change since it's error right now, and debug is commonly not stored in production. 
But if you think 'debug' the better level, I think that's definitely fine by me 👍

**Comment by peterklijn** [comment]
Fixed in https://github.com/zalando/skipper/pull/1659
Released in [version 0.13.3](https://github.com/zalando/skipper/releases/tag/v0.13.3)
